2004-04-13  James E Wilson  <wilson@specifixinc.com>

	* c-opt.c (c_common_post_options): If this_input_filename is NULL,
	increment errorcount and return false instead of true.

--- gcc/c-opts.c.jj	2004-10-18 15:16:08.000000000 +0200
+++ gcc/c-opts.c	2004-11-05 16:49:48.307731729 +0100
@@ -1176,8 +1176,12 @@ c_common_post_options (const char **pfil
 
   *pfilename = this_input_filename
     = cpp_read_main_file (parse_in, in_fnames[0]);
+  /* Don't do any compilation or preprocessing if there is no input file.  */
   if (this_input_filename == NULL)
-    return true;
+    {
+      errorcount++;
+      return false;
+    }
 
   if (flag_working_directory
       && flag_preprocess_only && ! flag_no_line_commands)
